pkg/cloud/services/networking/client.go

```diff
@@ -0,0 +1,327 @@
+	mc := metrics.NewMetricPrometheusContext("server_os_interface", "delete")
+	interfaceInfo, err := routers.RemoveInterface(c.serviceClient, id, opts).Extract()
+	if mc.ObserveRequestIgnoreNotFound(err) != nil {
+		return nil, err
+	}
+	return interfaceInfo, nil
+}
+
+func (c networkClient) ReplaceAllAttributesTags(resourceType string, resourceID string, opts attributestags.ReplaceAllOptsBuilder) ([]string, error) {
+	return attributestags.ReplaceAll(c.serviceClient, resourceType, resourceID, opts).Extract()
+}
+
+func (c networkClient) ListRouter(opts routers.ListOpts) ([]routers.Router, error) {
+	mc := metrics.NewMetricPrometheusContext("router", "list")
+	allPages, err := routers.List(c.serviceClient, opts).AllPages()
+	if mc.ObserveRequest(err) != nil {
+		return nil, err
+	}
+	return routers.ExtractRouters(allPages)
+}
+
+func (c networkClient) ListFloatingIP(opts floatingips.ListOptsBuilder) ([]floatingips.FloatingIP, error) {
+	mc := metrics.NewMetricPrometheusContext("floating_ip", "list")
+	allPages, err := floatingips.List(c.serviceClient, opts).AllPages()
+	if mc.ObserveRequest(err) != nil {
+		return nil, err
+	}
+	return floatingips.ExtractFloatingIPs(allPages)
+}
+
+func (c networkClient) CreateFloatingIP(opts floatingips.CreateOptsBuilder) (*floatingips.FloatingIP, error) {
+	mc := metrics.NewMetricPrometheusContext("floating_ip", "create")
+	fip, err := floatingips.Create(c.serviceClient, opts).Extract()
+	if mc.ObserveRequest(err) != nil {
+		return nil, err
+	}
+	return fip, nil
+}
+
+func (c networkClient) DeleteFloatingIP(id string) error {
+	mc := metrics.NewMetricPrometheusContext("floating_ip", "delete")
+	return mc.ObserveRequest(floatingips.Delete(c.serviceClient, id).ExtractErr())
+}
+
+func (c networkClient) GetFloatingIP(id string) (*floatingips.FloatingIP, error) {
+	mc := metrics.NewMetricPrometheusContext("floating_ip", "list")
+	fip, err := floatingips.Get(c.serviceClient, id).Extract()
+	if mc.ObserveRequest(err) != nil {
+		return nil, err
+	}
+	return fip, nil
+}
+
+func (c networkClient) UpdateFloatingIP(id string, opts floatingips.UpdateOptsBuilder) (*floatingips.FloatingIP, error) {
+	mc := metrics.NewMetricPrometheusContext("floating_ip", "update")
+	fip, err := floatingips.Update(c.serviceClient, id, opts).Extract()
+	if mc.ObserveRequest(err) != nil {
+		return nil, err
+	}
+	return fip, nil
+}
+
+func (c networkClient) ListPort(opts ports.ListOptsBuilder) ([]ports.Port, error) {
+	mc := metrics.NewMetricPrometheusContext("port", "list")
+	allPages, err := ports.List(c.serviceClient, opts).AllPages()
+	if mc.ObserveRequest(err) != nil {
+		return nil, err
+	}
+	return ports.ExtractPorts(allPages)
+}
+
+func (c networkClient) CreatePort(opts ports.CreateOptsBuilder) (*ports.Port, error) {
+	return ports.Create(c.serviceClient, opts).Extract()
+}
+
+func (c networkClient) DeletePort(id string) error {
```

Comment:
I'd suggest we consider add metrics for those kind of API call as well , but no need to be in this PR as this is a pure refactory